presto-main/src/main/java/com/facebook/presto/spiller/BinaryFileSingleStreamSpiller.java

```diff
@@ -99,7 +107,8 @@ private void checkNoSpillInProgress()
 
     private void writePages(Iterator<Page> pageIterator)
     {
-        try (SliceOutput output = new OutputStreamSliceOutput(new FileOutputStream(targetFileName.toFile(), true))) {
+        try (SliceOutput output = new OutputStreamSliceOutput(new FileOutputStream(targetFileName.toFile(), true), BUFFER_SIZE)) {
+            memoryContext.setBytes(BUFFER_SIZE);
```

Comment:
4k of memory is not a lot, so even if you are creating 10k of these you have only used 40MB of memory.  Do you really thing we should bother accounting for this?